code/datums/components/hazard_area.dm

```diff
@@ -0,0 +1,72 @@
+/**
+ * This is a relatively simple component that attempts to deterr the parent of the component away
+ * from a specific area or areas. By default it simply applies a penalty where all movement is
+ * four times slower than usual and any action that would affect your 'next move' has a penalty
+ * multiplier of 4 attached.
+ */
+/datum/component/hazard_area
+	/// The blacklist of areas that the parent will be penalized for entering
+	var/list/area_blacklist
+	/// The whitelist of areas that the parent is allowed to be in. If set this overrides the blacklist
+	var/list/area_whitelist
+	/// A variable storing the typepath of the last checked area to prevent any further logic running if it has not changed
+	VAR_PRIVATE/last_parent_area
+
+/datum/component/hazard_area/Initialize(area_blacklist, area_whitelist)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(!islist(area_blacklist) && !islist(area_whitelist))
+		stack_trace("[type] - neither area_blacklist nor area_whitelist were provided.")
+		return COMPONENT_INCOMPATIBLE
+	src.area_blacklist = area_blacklist
+	src.area_whitelist = area_whitelist
+	RegisterSignal(parent, COMSIG_MOB_CLIENT_MOVED, .proc/handle_parent_move)
+
+/datum/component/hazard_area/Destroy(force, silent)
+	. = ..()
+	UnregisterSignal(parent, COMSIG_MOB_CLIENT_MOVED)
```

Comment:
this is probably better done in `UnregisterFromParent`

Reply:
Not sure why you resolved this, it's correct.

Anything to do with parent should be in the parent proc calls.